lib/hanami/cli/commands/app/routes.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require "hanami"
+require "hanami/router/inspector"
+
+module Hanami
+  module CLI
+    module Commands
+      module App
+        # Inspect the application and slice routes
+        #
+        # Without the `slice` option, it will inspect the application routes:
+        #
+        # ```
+        # $ hanami routes
+        # ```
+        #
+        # Otherwise, provide the slice name:
+        #
+        # ```
+        # $ hanami routes --slice=main
+        # ```
+        #
+        # All the formatters available from `hanami-router` are available:
+        #
+        # ```
+        # $ hanami routes --format=csv
+        # ```
+        #
+        # You can also use a custom formatter registered in the application
+        # container. You can identify it by its key:
+        #
+        # ```
+        # $ hanami routes --format=custom_routes_formatter
+        # ```
+        class Routes < Hanami::CLI::Command
+          desc "Inspect application and slice routes"
+
+          option :format,
+                 default: :human_friendly,
+                 required: false,
+                 desc: "Output format: one supported by hanami-router (like csv) or the key for a custom registered formatter"
+
+          option :slice,
+                 required: false,
+                 desc: "Slice of the routes to print. Defaults to `nil`, which matches the app slice."
+
+          DEFAULT_FORMAT = "human_friendly"
+
+          # @api private
+          def call(format: DEFAULT_FORMAT, slice: nil)
+            require "hanami/prepare"
+            inspector = Hanami::Router::Inspector.new(formatter: resolve_formatter(format))
+            resolve_slice(slice).router(inspector: inspector)
+            out.puts inspector.call
+          end
+
+          private
+
+          def resolve_formatter(format)
+            resolve_formatter_from_hanami_router(format) ||
+              resolve_formatter_from_app(format)
+          end
+
+          def resolve_formatter_from_hanami_router(format)
+            file = "hanami/router/formatter/#{format}"
+            require file
+
+            file
+              .then { |f| inflector.classify(f) }
+              .then { |class_name| inflector.constantize(class_name) }
+              .then(&:new)
+          rescue LoadError
+            false
+          end
+
+          def resolve_formatter_from_app(format)
+            app[format]
+          end
+
+          def resolve_slice(slice)
+            slice ? app.slices[slice.to_sym] : app
+          end
+
+          def app
```

Comment:
@waiting-for-dev Could you please move this to `#initialize`?

```ruby
def initialize(app: Hanami.app, **)
  super
  @app = app
end

private

attr_reader :app
```

Reply:
@jodosha, at that point, `Hanami.app` isn't prepared yet, so [it'll raise an error](https://github.com/hanami/hanami/blob/70eed47749e5df602f5749d77dbc529d431f5602/lib/hanami.rb#L15).